plugins/com.python.pydev.analysis/src/com/python/pydev/analysis/AnalysisPreferences.java

```diff
@@ -217,32 +217,28 @@ public static boolean doAutoImport() {
             return TESTS_DO_AUTO_IMPORT;
         }
 
-        return PydevPrefs.getAnalysisEclipsePreferences().getBoolean(
-                AnalysisPreferenceInitializer.DO_AUTO_IMPORT,
-                AnalysisPreferenceInitializer.DEFAULT_DO_AUT_IMPORT);
+        return PyAnalysisScopedPreferences.getBoolean(AnalysisPreferenceInitializer.DO_AUTO_IMPORT, null);
```

Comment:
I think you should change the `doAutoImport` to receive the project adaptable... in the places where the completion comes from the IScriptConsoleViewer it can be null, but on the cases where there's a `CompletionRequest` the project is available through: `request.nature != null ? request.nature.getProject(): null`.

Reply:
As with the CtxParticipant, I didn't realize the nature carries a reference to the project it belongs to. Thanks for pointing that out.